panda-rs/src/api/utils.rs

```diff
@@ -0,0 +1,33 @@
+// TODO: cannot live in panda-macros due to arch feature flags?
+
+#[macro_export]
+#[cfg(any(feature = "i386", feature = "x86_64"))]
+macro_rules! cpu_arch_state {
```

Comment:
THIS IS SUCH A GOOD ADDITION. not sure how this slipped my mind tbh. One minor nitpick. You could probably make a type alias for the `CPU[arch]State` type so that you don't have to duplicate macro definitions

I suppose the structures themselves could also use a type alias but I can't think of a good way to handle that? So for now macro + cfg-gated type alias might be best

Reply:
Ah, that's a perfect usecase for type aliasing! Will do.

I don't know if structures themselves would work since `bindgen` types `env_ptr` as `*mut c_void` (correct, same as the C code), so an alias might force all void pointers to type as the arch CPU unless there's some way to scope? Anyhow macro + cfg-gated type alias seems really solid.